neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/common/dialect/impl/RemoteNeo4jAssociationSnapshot.java

```diff
@@ -29,6 +31,11 @@ public RemoteNeo4jAssociationSnapshot(Map<RowKey, Tuple> tuples) {
 	@Override
 	public Tuple get(RowKey rowKey) {
 		Tuple tuple = tuples.get( rowKey );
+		for ( Entry<RowKey, Tuple> entry : tuples.entrySet() ) {
+			if ( RemoteNeo4jHelper.matches( entry.getKey(), rowKey ) ) {
+				return entry.getValue();
+			}
+		}
```

Comment:
As you said, this is more a mistake. I think it's something left from before. I'll keep it as a fall-back